ykrt/src/mt.rs

```diff
@@ -434,16 +341,21 @@ impl MT {
 
     /// Add a compilation job for `sir` to the global work queue.
     fn queue_compile_job(&self, trace: IRTrace, mtx: Arc<Mutex<Option<Box<CompiledTrace>>>>) {
-        let f = Box::new(move || {
-            // FIXME compile IR trace to executable code here.
-            let compiled = todo!();
-            *mtx.lock() = Some(Box::new(compiled));
-            // FIXME: although we've now put the compiled trace into the mutex, there's no
-            // guarantee that the Location for which we're compiling will ever be executed
-            // again. In such a case, the memory has, in essence, leaked.
-        });
-
-        self.queue_job(f);
+        // FIXME: Until we can find a way of testing non-deterministic and time-sensitive JIT
```

Comment:
I'm decidedly unkeen on this change because you're commenting out very critical code that will now go untested.

I'd actually prefer to cheat and let compilation carry in a thread but wait for it after the `let f = ...` stuff with something like:

```
    loop {
            match loc.try_lock() {
                Ok(x) => ls = x,
                Err(_) => todo!()
            }
            let hl = unsafe { ls.hot_location() };
            let hl_ptr = hl as *mut _ as *mut ();
            match hl {
                  HotLocation::Compiled(_) = break;
                  _ => ()
            }
    }
```

Warning: untested!

Reply:
Did you mean `mtx.lock()`?

Reply:
(instead of `loc.try_lock()`)

I'm a bit confused, as `loc` doesn't exist.

Reply:
`ls` is also undefined.

Reply:
Erp, true, that stuff is defined in the main lock.

I think the easiest thing is to get this PR in and for me to do something in a subsequent PR.

Reply:
Left as-is then.